Parsers/ASimAuthentication/Parsers/vimAuthenticationBarracudaWAF.yaml

```diff
@@ -0,0 +1,111 @@
+Parser:
+  Title: ASIM Authentication parser for Barracuda WAF
+  Version: '0.1'
+  LastUpdated: Jun 13 2023
+Product:
+  Name: Barracuda WAF
+Normalization:
+  Schema: Authentication
+  Version: '0.1.3'
+References:
+- Title: ASIM Authentication Schema
+  Link: https://aka.ms/ASimAuthenticationDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: Logs type Description in Barracuda WAF
+  Link: https://campus.barracuda.com/product/webapplicationfirewall/doc/78810008/types-of-logs/
+Description: |
+  This ASIM parser supports normalizing the Barracuda WAF logs to the ASIM authentication normalized schema.
+ParserName: vimAuthenticationBarracudaWAF
+EquivalentBuiltInParser: _Im_Authentication_BarracudaWAF
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: targetusername_has
+    Type: string
+    Default: '*'
+ParserQuery: |
+  let SeverityLookup = datatable (severity: int, EventSeverity: string)
+      [
+      0, "High", 
+      1, "High", 
+      2, "High", 
+      3, "Medium",
+      4, "Low",
+      5, "Low", 
+      6, "Informational",
+      7, "Informational" 
+  ];
+  let EventTypeLookup = datatable (
+      EventName_s: string,
+      EventType: string,
+      EventResult: string
+  )
+      [
+      "LOGIN", "Logon", "Success",
+      "UNSUCCESSFUL_LOGIN", "Logoff", "Failure",
+      "LOGOUT", "Logoff", "Success"
+  ];
+  let parser=(
+      starttime: datetime = datetime(null),
+      endtime: datetime = datetime(null),
+      targetusername_has: string = "*",
+      disabled: bool = false) {
+      barracuda_CL
+      | where not(disabled)
+          and (LogType_s == "AUDIT")
+          and (EventName_s in ("LOGIN", "LOGOUT", "UNSUCCESSFUL_LOGIN"))
```

Comment:
Column_ifexists is missing for all the fields here.

Reply:
Instead of column_ifexists we used the the approach suggested by you in mail to create one data table with the default values.